flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/testutils/KafkaMultipleTopicExternalContext.java

```diff
@@ -0,0 +1,125 @@
+ * splits.
+ */
+public class KafkaMultipleTopicExternalContext extends KafkaSingleTopicExternalContext {
+
+    private int numTopics = 0;
+
+    private final String topicPattern;
+
+    private final Map<String, SourceSplitDataWriter<String>> topicNameToSplitWriters =
+            new HashMap<>();
+
+    public KafkaMultipleTopicExternalContext(String bootstrapServers) {
+        super(bootstrapServers);
+        this.topicPattern =
+                "kafka-multiple-topic-[0-9]+-"
+                        + ThreadLocalRandom.current().nextLong(Long.MAX_VALUE);
+    }
+
+    @Override
+    public SourceSplitDataWriter<String> createSourceSplitDataWriter() {
+        String topicName = getTopicName();
+        createTopic(topicName, 1, (short) 1);
+        final KafkaPartitionDataWriter splitWriter =
+                new KafkaPartitionDataWriter(
+                        getKafkaProducerProperties(numTopics), new TopicPartition(topicName, 0));
+        topicNameToSplitWriters.put(topicName, splitWriter);
+        numTopics++;
+        return splitWriter;
+    }
+
+    @Override
+    public Source<String, ?, ?> createSource(Boundedness boundedness) {
+        KafkaSourceBuilder<String> builder = KafkaSource.builder();
+
+        if (boundedness == Boundedness.BOUNDED) {
+            builder = builder.setBounded(OffsetsInitializer.latest());
+        }
+
+        return builder.setGroupId("flink-kafka-multiple-topic-test")
+                .setBootstrapServers(bootstrapServers)
+                .setTopicPattern(Pattern.compile(topicPattern))
+                .setDeserializer(
+                        KafkaRecordDeserializationSchema.valueOnly(StringDeserializer.class))
+                .build();
+    }
+
+    @Override
+    public void close() {
+        topicNameToSplitWriters.forEach(
+                (topicName, splitWriter) -> {
+                    try {
+                        splitWriter.close();
+                        deleteTopic(topicName);
+                    } catch (Exception e) {
+                        kafkaAdminClient.close();
+                        throw new RuntimeException("Cannot close split writer", e);
+                    }
+                });
+        topicNameToSplitWriters.clear();
+        kafkaAdminClient.close();
+    }
+
+    private String getTopicName() {
+        return topicPattern.replace("[0-9]+", String.valueOf(numTopics));
+    }
+
+    @Override
+    public String toString() {
+        return "Multiple-topics Kafka";
+    }
+
+    /** . */
+    public static class Factory implements ExternalContext.Factory<String> {
+
+        private final String bootstrapServers;
+
+        public Factory(String bootstrapServers) {
+            this.bootstrapServers = bootstrapServers;
+        }
+
+        @Override
+        public ExternalContext<String> createExternalContext() {
+            return new KafkaMultipleTopicExternalContext(bootstrapServers);
```

Comment:
Here we should extract the bootstrap server. (I'm assuming now that the external resources are running or we can easily force it while calling this factory)